GNUmakefile.llvm

```diff
@@ -270,15 +270,20 @@ endif
 ifneq "$(LLVM_CONFIG)" ""
   CLANG_CFL += -I$(shell dirname $(LLVM_CONFIG))/../include
 endif
-CLANG_CPPFL  = `$(LLVM_CONFIG) --cxxflags` -fno-rtti -fPIC $(CXXFLAGS) -Wno-deprecated-declarations
-CLANG_LFL    = `$(LLVM_CONFIG) --ldflags` $(LDFLAGS)
+CLANG_CPPFL  =  $(shell $(LLVM_CONFIG) --cxxflags) -fPIC $(CXXFLAGS)
+CLANG_LFL    =  $(shell $(LLVM_CONFIG) --ldflags --libs passes --system-libs) $(LDFLAGS)
 
+ifeq "$(shell $(LLVM_CONFIG) --has-rtti) 2> /dev/null" "YES"
+  CLANG_CPPFL += -frtti
```

Comment:
why do you add rtti? I have never experienced it being needed.

Reply:
While the LLVM project "[does not use exceptions or RTTI]", [LLVM_ENABLE_RTTI] is still a valid build option which would require `-frtti` as a codegen option for compiling the AFL++ passes as "[Mixing code compiled with -frtti with that compiled with -fno-rtti may not work]".  The other branch of this conditional should almost always be hit with most LLVM binary configurations in the wild, this is a safety check for non-default configurations.

I tested LLVM configs with both rtti and no rtti and while the Makefile without this patch worked, `llvm-config` exposes this option for good cause in preventing bugs and it should be leveraged in building plugins to its infrastructure. The [Boost] project also warn of mixing such code. 

@vanhauser-thc Did you test with a LLVM binary configured with RTTI Enabled?

[does not use exceptions or RTTI]: https://github.com/llvm/llvm-project/blob/llvmorg-12.0.0-rc1/llvm/docs/CodingStandards.rst#do-not-use-rtti-or-exceptions

[LLVM_ENABLE_RTTI]: https://github.com/llvm/llvm-project/blob/llvmorg-12.0.0-rc1/llvm/docs/CMake.rst#llvm-specific-variables

[Mixing code compiled with -frtti with that compiled with -fno-rtti may not work]: https://gcc.gnu.org/onlinedocs/gcc/C_002b_002b-Dialect-Options.html

[Boost]: https://www.boost.org/doc/libs/1_75_0/doc/html/boost_typeindex/mixing_sources_with_rtti_on_and_.html

Reply:
passes are just their own .cc and afl-llvm-common.c, compiled together with the same compiler with the same options at the same time.
instrumenting a binary with a pass does not compile that pass into the target.
hence rtti or not has zero impact. it is just not needed. I dont mind if it is there, it just doesnt make a difference.

Reply:
I understand that pass code and target code are not linked together. However, pass code and LLVM / Clang code are linked when the pass is invoked. The `CLANG_CPPFL` variable compiles AFL++ code which will be linked against LLVM / Clang code, correct? If so then, the `rtti` of the code should then match how the LLVM / Clang was compiled (i.e. `llvm-config --has-rtti`)

Reply:
ah right, that makes sense!